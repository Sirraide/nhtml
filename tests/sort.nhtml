#children {
    div (a)
    div (c)
    div (f)
    div (b)
}

eval {
    function lex_compare(a, b)  {
        let sz = Math.min(a.length, b.length)
        for (let i = 0; i < sz; ++i) {
            if (a.charCodeAt(i) < b.charCodeAt(i)) return -1
            if (a.charCodeAt(i) > b.charCodeAt(i)) return 1
        }
        return a.length - b.length
    }

    let chs = $('#children')
    let divs = [...chs.children].sort((a, b) => lex_compare(a.text, b.text))
    chs.children = divs
}