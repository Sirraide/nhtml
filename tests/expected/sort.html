<div id='children'>
    <div>a</div>
    <div>b</div>
    <div>c</div>
    <div>f</div>
</div>
